packages/starlight/__tests__/test-config.ts

```diff
@@ -0,0 +1,16 @@
+/// <reference types="vitest" />
+
+import { getViteConfig } from 'astro/config';
+import type { z } from 'astro/zod';
+import { vitePluginStarlightUserConfig } from '../integrations/virtual-user-config';
+import { StarlightConfigSchema } from '../utils/user-config';
+
+export function defineVitestConfig(config: z.input<typeof StarlightConfigSchema>) {
+  return getViteConfig({
+    plugins: [
+      vitePluginStarlightUserConfig(StarlightConfigSchema.parse(config), {
+        root: new URL(import.meta.url),
+      }),
+    ],
+  });
+}
```

Comment:
This config helper wraps Astro’s `getViteConfig` to set virtual modules in the test environment using Starlight’s Vite plugin. If each test suite could contain an `astro.config` instead, we could perhaps use the Starlight integration directly instead.